### ISSUE TITLE
the learning rate can not change?

### ISSUE THREAD

**Issue by xuweidongkobe** [MhLiao/DB#101]
the initial lr is 0.007, I change the lr to 0.001 in my *yaml，but in train log the lr still 0.007.
then I input the --lr 0.001, but in train log the lr still 0.007. 
Can you tell me how to solve this problem? Thanks

**Comment by deepseek** [comment]
@xuweidongkobe I faced the same issue, the solution is to change the value in [learning_rate.py](https://github.com/MhLiao/DB/blob/becccdfb3b979e86f83a289beef8206cd75810b1/training/learning_rate.py#L90)
close this issue

**Comment by xuweidongkobe** [comment]
@deepseek thank you, it can works. But I want to know why change the *yaml file does not work?

**Comment by JimmyJuan** [comment]
@xuweidongkobe the class DecayLearningRate in https://github.com/MhLiao/DB/blob/becccdfb3b979e86f83a289beef8206cd75810b1/training/learning_rate.py#L90 reload learning rate again.
Since the class is inherited from "Configurable", it can be configured in the yaml file rather than to modify source code:
in the yaml file
`
            learning_rate:
                class: DecayLearningRate
                lr: 0.001
                epochs: 1200
`

then it works

**Comment by zengchan** [comment]
I think you can add a prepare function in DecayLearningRate like the following BuiltinLearningRate to obtain learning rate from optimizer args.
